pkg/util/stringutils.go

```diff
@@ -3,25 +3,27 @@ package util
 import (
 	"crypto/sha512"
 	"regexp"
-	"strconv"
 	"strings"
 )
 
-var (
-	reTrue = regexp.MustCompile("(?i)^true|yes|1$")
-)
-
+// Sanitize ensures a string only contains alphanumeric characters, and starts
+// with a letter
 func Sanitize(str string) string {
 	str = strings.ToLower(str)
 	// replace all non-alphanumeric characters by "-"
 	str = regexp.MustCompile("[^a-z0-9-]").ReplaceAllString(str, "-")
 	// replace multiple subsequent "-" by a single "-"
 	str = regexp.MustCompile("-+").ReplaceAllString(str, "-")
 	// remove "-" and numbers at beginning, "-" at end of string
-	str = regexp.MustCompile("^[0-9-]*([^-]*)-?$").ReplaceAllString(str, "${1}")
+	str = regexp.MustCompile("^[0-9-]+|-+$").ReplaceAllString(str, "")
 	return str
 }
 
+// SanitizeWithMinLength applies `Sanitize`, and then ensures the result is at least `minLength` characters long.
+//
+// If the resulting string is too short, the SHA-512 hash of the original input
+// will be taken, and the first `minLength` characters of its hex-encoding will
```

Comment:
Note that ByteArrayToAlpha does NOT do hex encoding, instead of 0-f it uses a-p, so we better don't call it hex encoding here.